### PR TITLE
Add NpcCompositionChanged event and fix NPC Indicators when NPC composition changes

### LINKED ISSUE #5161: NPC indicator highlighting after NPC transforming
Currently if a tagged (from the name in settings) NPC spawns it will be highlighted. If it then proceeds to transform into another NPC which is _not_ highlighted, it will stay tagged because of its initial spawn.

This is mainly an issue in the Theatre of Blood's Nylocas room.

Expected behavior: the NPC would no longer be highligted after transforming into another NPC.

### PR DESCRIPTION
**Pull request by WooxSolo**
Fixes #5161

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java [hunk 1]
```diff
@@ -311,6 +312,29 @@ public void onNpcSpawned(NpcSpawned npcSpawned)
 		}
 	}
 
+	@Subscribe
+	public void onNpcCompositionChanged(NpcCompositionChanged event)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java]: You should probably make common method for body of this event and use it in both npcspawned and here

File: runelite-mixins/src/main/java/net/runelite/mixins/RSNPCMixin.java [hunk 2]
```diff
@@ -110,6 +111,13 @@ public void onCompositionChanged(RSNPCComposition composition)
 		}
 	}
 
+	@FieldHook("composition")
+	@Inject
+	public void afterCompositionChanged(int idx)
+	{
+		client.getCallbacks().post(new NpcCompositionChanged(this));
```
Comment (Adam-) [inline runelite-mixins/src/main/java/net/runelite/mixins/RSNPCMixin.java]: new assignment can be null on npc despawn

**Comment by Adam-** [comment]
FYI this depends on !34

File: runelite-mixins/src/main/java/net/runelite/mixins/RSNPCMixin.java (hunk 2 above)
Comment (WooxSolo) [reply runelite-mixins/src/main/java/net/runelite/mixins/RSNPCMixin.java]: I don't understand why this would be a problem

Comment (Adam-) [reply runelite-mixins/src/main/java/net/runelite/mixins/RSNPCMixin.java]: It fires the event for comp change, when it is really a npc despawn. Seems bad?

Comment (WooxSolo) [reply runelite-mixins/src/main/java/net/runelite/mixins/RSNPCMixin.java]: Well technically the composition does change when it despawns. I don't have a strong opinion about it though so I guess I'll change it

**Comment by Adam-** [comment]
the event part of this was added in caaf98a8597a4be0a202c49256d083e8111a8b34

**Pull request closed without merging**